src/Modules/OrchardCore.Commerce/Services/ProductAttributeProvider.cs

```diff
@@ -28,9 +29,18 @@ public class ProductAttributeProvider : IProductAttributeProvider
 
         var name = partDefinition.Name + "." + attributeFieldDefinition.Name;
 
-        var element = value is JsonElement jsonElement
-            ? jsonElement
-            : JsonDocument.Parse(value.ToString()!).RootElement;
+        var element = default(JsonElement);
+        element = value switch
+        {
+            JsonElement jsonElement => jsonElement,
+
+            // String needs to be lowercase for JSON parsing to work.
+#pragma warning disable CA1308 // Normalize strings to uppercase
+            JValue jValue => JsonDocument.Parse(jValue.Value.ToString().ToLowerInvariant()).RootElement,
+#pragma warning restore CA1308 // Normalize strings to uppercase
```

Comment:
Normalizing to uppercase doesn't work in this case:
![devenv_LyrfBdXAa3](https://github.com/OrchardCMS/OrchardCore.Commerce/assets/51411356/6eb195ca-13c5-4d59-8d50-7e0af616a01e)